### PR TITLE
Remove orphan lock file

### PR DESCRIPTION
**Pull request by tamilmani1989**
**What this PR does / why we need it**:
This PR make sure that  CNI process is not blocked bacause of orphaned lock file. The cni will remove lock file if the process that is holding lockfile exited

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: cni/ipam/plugin/main.go [hunk 1]
```diff
@@ -28,11 +28,19 @@ func main() {
 
 	if err := ipamPlugin.Plugin.InitializeKeyValueStore(&config); err != nil {
 		fmt.Printf("Failed to initialize key-value store of ipam plugin, err:%v.\n", err)
+
+		if ipamPlugin.Plugin.IsSafeToRemoveLock(ipamPlugin.Plugin.Name) {
```
Comment (jaer-tsun) [inline cni/ipam/plugin/main.go]: duplicate code, maybe make this a func

File: cni/plugin.go [hunk 2]
```diff
@@ -186,9 +188,9 @@ func (plugin *Plugin) InitializeKeyValueStore(config *common.PluginConfig) error
 }
 
 // Uninitialize key-value store
-func (plugin *Plugin) UninitializeKeyValueStore() error {
+func (plugin *Plugin) UninitializeKeyValueStore(forceUnlock bool) error {
```
Comment (jaer-tsun) [inline cni/plugin.go]: force is good enough

File: cni/plugin.go [hunk 3]
```diff
@@ -198,3 +200,37 @@ func (plugin *Plugin) UninitializeKeyValueStore() error {
...
+			return false
+		}
+
+		if len(content) <= 0 {
+			log.Errorf("Num bytes read from lock file is 0")
+			return false
+		}
+
+		pid := string(content)
+		pid = strings.Trim(pid, "\n")
```
Comment (jaer-tsun) [inline cni/plugin.go]: pid := strings.Trim(string(content), "\n") seems cleaner

File: cni/plugin.go [hunk 4]
```diff
@@ -198,3 +200,37 @@ func (plugin *Plugin) UninitializeKeyValueStore() error {
 
 	return nil
 }
+
+// check if safe to remove lockfile
+func (plugin *Plugin) IsSafeToRemoveLock(processName string) bool {
+	if plugin.Store != nil {
```
Comment (jaer-tsun) [inline cni/plugin.go]: you may need to check if plugin is nil as well

File: cni/ipam/plugin/main.go (hunk 1 above)
Comment (tamilmani1989) [reply cni/ipam/plugin/main.go]: cannot do this..will explain offline

**Pull request closed without merging**